apps/src/music/progress/MusicValidator.ts

```diff
@@ -0,0 +1,73 @@
+// Music Lab specific validations.
+
+import MusicPlayer from '../player/MusicPlayer';
+import ConditionsChecker, {KnownConditions} from './ConditionsChecker';
+import {PlaybackEvent} from '../player/interfaces/PlaybackEvent';
+import {Validator} from './ProgressManager';
+
+const KnownConditions: KnownConditions = {
+  PLAYED_ONE_SOUND: 'played_one_sound',
+  PLAYED_TWO_SOUNDS_TOGETHER: 'played_two_sounds_together',
+  PLAYED_THREE_SOUNDS_TOGETHER: 'played_three_sounds_together'
+};
+
+export default class MusicValidator extends Validator {
+  private player: MusicPlayer;
+  private getIsPlaying: Function;
+  private conditionsChecker: ConditionsChecker;
+
+  constructor(getIsPlaying: Function, player: MusicPlayer) {
```

Comment:
nit: type of getIsPlaying can be `: () => boolean`

Reply:
Nice.